pkg/policy/api/rule_validation.go

```diff
@@ -641,15 +641,20 @@ func (pr *PortDenyRule) sanitize() error {
 
 func (pp *PortProtocol) sanitize(hasDNSRules bool) (isZero bool, err error) {
 	if pp.Port == "" {
-		return isZero, errors.New("Port must be specified")
+		if !option.Config.EnableUnknownProtocols {
```

Comment:
as discussed, I think it's OK to allow empty ports for all protocols. The PolicyMap supports it just fine.

Reply:
I added the conditional validation per your review comment - https://github.com/cilium/cilium/pull/39872#discussion_r2144511283.